src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala

```diff
@@ -0,0 +1,183 @@
+package temple.generate.server.go.service.main
+
+import temple.ast.AttributeType.DateTimeType
+import temple.ast.{Annotation, AttributeType}
+import temple.generate.CRUD._
+import temple.generate.server.{CreatedByAttribute, ServiceRoot}
+import temple.generate.utils.CodeTerm.{CodeWrap, mkCode}
+import temple.generate.server.go.common.GoCommonGenerator._
+import temple.utils.StringUtils.doubleQuote
+
+import scala.Option.when
+import scala.collection.immutable.ListMap
+
+object GoServiceMainHandlersGenerator {
+
+  /** Generate a map for converting the fields of the DAO response to the JSON response  */
+  private def generateResponseMap(root: ServiceRoot): ListMap[String, String] =
+    // Includes ID attribute and all attributes without the @server annotation
+    ListMap(root.idAttribute.name.toUpperCase -> s"${root.name}.${root.idAttribute.name.toUpperCase}") ++
+    root.attributes.collect {
+      case (name, attribute) if !attribute.accessAnnotation.contains(Annotation.Server) =>
+        (
+          name.capitalize,
+          s"${root.name}.${name.capitalize}" +
+          // Must add formatting to attributes with datetime type
+          (attribute.attributeType match {
+            case DateTimeType => s".${genFunctionCall("Format", "time.RFC3339")}"
+            case _            => ""
+          }),
+        )
+    }
+
+  /** Generate a handler method declaration */
+  private def generateHandlerDecl(root: ServiceRoot, operation: CRUD): String =
+    s"func (env *env) ${operation.toString.toLowerCase}${root.name.capitalize}Handler(w http.ResponseWriter, r *http.Request)"
+
+  /** Generate a errMsg declaration and http.Error call */
+  private def generateHTTPError(statusCodeEnum: String, errMsg: String, errMsgArgs: String*): String = {
+    val createErrorJSONargs =
+      if (errMsgArgs.nonEmpty) genMethodCall("fmt", "Sprintf", (doubleQuote(errMsg) +: errMsgArgs): _*)
+      else doubleQuote(errMsg)
+
+    mkCode.lines(
+      genDeclareAndAssign(
+        genMethodCall(
+          "util",
+          "CreateErrorJSON",
+          createErrorJSONargs,
+        ),
+        "errMsg",
+      ),
+      genMethodCall("http", "Error", "w", "errMsg", s"http.$statusCodeEnum"),
+    )
+  }
+
+  /** Generate the block for extracting an AuthID from the request header */
+  private def generateExtractAuthBlock(): String =
+    mkCode.lines(
+      genDeclareAndAssign(genMethodCall("util", "ExtractAuthIDFromRequest", "r.Header"), "auth", "err"),
+      genIfErr(
+        mkCode.lines(
+          generateHTTPError("StatusUnauthorized", "Could not authorize request: %s", genMethodCall("err", "Error")),
+          genReturn(),
+        ),
+      ),
+    )
+
+  /** Generate the list handler function */
+  private def generateListHandler(root: ServiceRoot, responseMap: ListMap[String, String]): String = {
```

Comment:
If they're all going to get this long, I feel like they should be broken into separate objects?